src/SDKs/Batch/DataPlane/Azure.Batch.IntegrationTests/InboundEndpointIntegrationTests.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace BatchClientIntegrationTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using BatchTestCommon;
+    using Fixtures;
+    using Microsoft.Azure.Batch;
+    using Microsoft.Azure.Batch.Common;
+    using IntegrationTestUtilities;
+    using Xunit;
+    using Xunit.Abstractions;
+
+    public class InboundEndpointIntegrationTests
+    {
+        private readonly ITestOutputHelper testOutputHelper;
+        private readonly PoolFixture poolFixture;
+        private static readonly TimeSpan TestTimeout = TimeSpan.FromMinutes(3);
+
+        public InboundEndpointIntegrationTests(ITestOutputHelper testOutputHelper)
+        {
+            this.testOutputHelper = testOutputHelper;
+        }
+
+        [Fact]
+        [Trait(TestTraits.Duration.TraitName, TestTraits.Duration.Values.ShortDuration)]
+        public void WhenPoolCreatedWithInboundEndpoints_EndpointsAreReturnedByPoolAndComputeNodes()
+        {
+            Action test = () =>
+            {
+                using (BatchClient batchCli = TestUtilities.OpenBatchClientAsync(TestUtilities.GetCredentialsFromEnvironment()).Result)
+                {
+                    string poolId = "InboundEndpoints-" + TestUtilities.GetMyName();
+
+                    try
+                    {
+                        var pool = CreatePool(batchCli, poolId);
+                        pool.NetworkConfiguration = GetNetworkConfiguration();
+                        pool.Commit();
+
+                        var actualPool = batchCli.PoolOperations.GetPool(poolId);
+
+                        var expectedInboundNatPools = pool.NetworkConfiguration.EndpointConfiguration.InboundNatPools.OrderBy(inp => inp.Name).ToList();
+                        var actualInboundNatPools = actualPool.NetworkConfiguration.EndpointConfiguration.InboundNatPools.OrderBy(inp => inp.Name).ToList();
+
+                        Assert.Equal(expectedInboundNatPools.Count, actualInboundNatPools.Count);
+
+                        for (var i = 0; i < expectedInboundNatPools.Count; i++)
+                        {
+                            ValidateEquality(expectedInboundNatPools[i], actualInboundNatPools[i]);
+
+                            if (expectedInboundNatPools[i].NetworkSecurityGroupRules != null && 
```

Comment:
Consider pulling this out into a helper method e.g. `{Validate|Assert}NetworkSecurityGroupRulesEqual`